File: 2706-buy-two-chocolates/2706-buy-two-chocolates.java
class Solution {
    public int buyChoco(int[] prices, int money) {
        int first = Integer.MAX_VALUE;
        int second = Integer.MAX_VALUE;

        for (int p : prices) {
            if (p < first) {
                second = first;
                first = p;
            } else if (p < second) {
                second = p;
            }
        }
        int sum = first + second;
        if (sum <= money) {
            return money - sum;
        }
        return money;
    }
}

File: 2706-buy-two-chocolates/2706-buy-two-chocolates.js
/**
 * @param {number[]} prices
 * @param {number} money
 * @return {number}
 */
var buyChoco = function (prices, money) {
    let first = Infinity;
    let second = Infinity;

    for (let i = 0; i < prices.length; i++) {
        if (prices[i] < first) {
            second = first;
            first = prices[i];
        } else if (prices[i] < second) {
            second = prices[i];
        }
    }

    let sum = first + second;

    if (sum <= money) {
        return money - sum;
    }

    return money;
};